### PR TITLE
[nGraph] Add documentation strings

### PR DESCRIPTION
**Pull request by jiwaszki**
### Details:
 - Added doc strings to nGraph classes.
 - Changed order of class registering in pybind. That allows automatic signatures generator (from pybind) to see classes in form `_pyngraph.class_name` instead of C++ notation in form of `ngraph::class_name`.
 - Added py::args to functions which suggest automatic signatures generator to replace names in signatures and allows users to specify the argument in a function call `func(argument_name = value)`.

### Tickets:
 - *51816*

### CODE REVIEW

**Review by tomdol — APPROVED** [review]

**Comment by ilyachur** [comment]
@ntyukaev FYI

**Review by mbencer — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Comment by ilyachur** [comment]
@postrational Could you have a look?

**Review by postrational — CHANGES_REQUESTED** [review]

File: ngraph/python/src/pyngraph/function.cpp [hunk 1]
```diff
@@ -19,24 +19,214 @@ void regclass_pyngraph_Function(py::module m)
...
+
+                    Returns
+                    ----------
+                    get_ops : List[Node]
+                        List of Nodes representing ops used in function.
+                 )");
+    function.def("get_ordered_ops",
+                 &ngraph::Function::get_ordered_ops,
+                 R"(
+                    Return ordered ops used in the function.
```
[the remark this answers is not in the document]
Comment (jiwaszki) [reply ngraph/python/src/pyngraph/function.cpp]: Done.

File: ngraph/python/src/pyngraph/function.cpp [hunk 2]
```diff
@@ -19,24 +19,214 @@ void regclass_pyngraph_Function(py::module m)
     function.doc() = "ngraph.impl.Function wraps ngraph::Function";
     function.def(py::init<const std::vector<std::shared_ptr<ngraph::Node>>&,
                           const std::vector<std::shared_ptr<ngraph::op::Parameter>>&,
-                          const std::string&>());
+                          const std::string&>(),
+                 py::arg("results"),
+                 py::arg("parameters"),
+                 py::arg("name"),
+                 R"(
+                    Create Function.
```
[the remark this answers is not in the document]
Comment (jiwaszki) [reply ngraph/python/src/pyngraph/function.cpp]: Done

File: ngraph/python/src/pyngraph/dimension.cpp [hunk 3]
```diff
@@ -39,14 +79,107 @@ void regclass_pyngraph_Dimension(py::module m)
...
+
+                Returns
+                ----------
+                get_length : int
+                    Value of the dimension.
+            )");
+    dim.def("get_min_length",
+            &ngraph::Dimension::get_min_length,
+            R"(
+                Convert this dimension's min_dimension to integer.
```
[the remark this answers is not in the document]
Comment (jiwaszki) [reply ngraph/python/src/pyngraph/dimension.cpp]: Done

File: ngraph/python/src/pyngraph/dimension.cpp [hunk 4]
```diff
@@ -39,14 +79,107 @@ void regclass_pyngraph_Dimension(py::module m)
...
+                Returns
+                ----------
+                relaxes : bool
+                    True if this dimension relaxes d, else False.
+            )");
+    dim.def("refines",
+            &ngraph::Dimension::refines,
+            py::arg("d"),
+            R"(
+                Check whether this dimension is a refinement of the argument.
```
[the remark this answers is not in the document]
Comment (jiwaszki) [reply ngraph/python/src/pyngraph/dimension.cpp:177]: Done

**Review by postrational — APPROVED** [review]

**Review by etusien — APPROVED** [review]

**Pull request merged**